backend/dataall/modules/s3_datasets_shares/services/share_managers/s3_access_point_share_manager.py

```diff
@@ -351,21 +351,6 @@ def grant_target_role_access_policy(self):
         if kms_key_id:
             kms_target_resources = [f'arn:aws:kms:{self.dataset_region}:{self.dataset_account_id}:key/{kms_key_id}']
 
-        managed_policy_exists = True if share_policy_service.get_managed_policies() else False
```

Comment:
Just to confirm my understanding. This code block is no longer needed because in `merge_statements_and_update_policies` we already create the IAM policy: `dataall.modules.s3_datasets_shares.services.s3_share_managed_policy_service.S3SharePolicyService._create_empty_policies_with_indexes`